examples/remote_data_manager.py

```diff
@@ -0,0 +1,56 @@
+"""
+========================================
+Using sunpy.data.manager
```

Comment:
This is an interesting new class of example, it's one which is primarily for developers of other libs rather than users? I feel like a new folder is in order I just don't know what to title it?

Reply:
> other libs rather than users? 

I wrote this with users in mind. If they had to write some code which used a remote file this would be useful. If they want to replace some file, this example would be useful.

Developers of other libs can use this as well as the developer docs.